### PR TITLE
Scarecrow rendering fixed, stop exception.

### PR DESCRIPTION
**Pull request by Raycoms**
Review please @Minecolonies/maintainers

Sometimes may not find the scarecrow in the world, in this case don't rotate.
Sometimes when placing a building or removing it, minecraft crashes since the list may be iteratet at that moment.

### CODE REVIEW

**Comment by Kostronor** [comment]
LGTM! Let's :shipit:

File: src/main/java/com/minecolonies/client/render/TileEntityScarecrowRenderer.java [hunk 1]
```diff
@@ -97,19 +89,24 @@ public void renderTileEntityAt(@NotNull ScarecrowTileEntity te, double posX, dou
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/client/render/TileEntityScarecrowRenderer.java]: The line above this, what happens if the TE had disappeared, now that we moved the check?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L657](https://github.com/Minecolonies/minecolonies/blob/7ec290d68188f4cef517c508dc98252333dd6b03/src/main/java/com/minecolonies/colony/Colony.java#L657): Remove this unused method parameter "oldSubscribers". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)
1. MAJOR [Colony.java#L659](https://github.com/Minecolonies/minecolonies/blob/7ec290d68188f4cef517c508dc98252333dd6b03/src/main/java/com/minecolonies/colony/Colony.java#L659): Add parentheses to make the operator precedence explicit. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)

File: src/main/java/com/minecolonies/util/RecipeHandler.java [hunk 2]
```diff
@@ -45,8 +45,8 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ItemStack(ModBlocks.blockSubstitution, 16), "XXX", "X#X", "XXX", 'X', Blocks.planks, '#', ModItems.scanTool);
         GameRegistry.addRecipe(new ItemStack(ModBlocks.blockHutFarmer, 1), "XXX", "X#X", "XXX", 'X', Blocks.planks, '#', Items.wooden_hoe);
         GameRegistry.addRecipe(new ItemStack(ModBlocks.blockHutFarmer, 2), "XXX", "X#X", "XXX", 'X', Blocks.planks, '#', Items.stone_hoe);
-        GameRegistry.addRecipe(new ItemStack(ModBlocks.blockHutField, 1), " Y ", "X#X", " X ", 'X', Items.stick, '#', Items.leather_chestplate, 'Y', Blocks.hay_block);
-        GameRegistry.addRecipe(new ItemStack(Blocks.web, 1), "XXX", "XXX", "XXX", 'X', Items.string);
+        GameRegistry.addRecipe(new ItemStack(ModBlocks.blockHutField, 1), " Y ", "X#X", " X ", 'X', Items.stick, '#', Items.leather, 'Y', Blocks.hay_block);
+        GameRegistry.addRecipe(new ItemStack(Blocks.web, 1), "X X", " X ", "X X", 'X', Items.string);
 
         // Disabled for now
         // GameRegistry.addRecipe(new ItemStack(ModBlocks.blockBarrel, 1), "P P", "P P", " S ", 'P', Blocks.planks, 'S', Blocks.wooden_slab);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/util/RecipeHandler.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L657](https://github.com/Minecolonies/minecolonies/blob/2c93afb9c00265d7894d87c94a983cd3b1ace2a6/src/main/java/com/minecolonies/colony/Colony.java#L657): Remove this unused method parameter "oldSubscribers". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)
1. MAJOR [Colony.java#L659](https://github.com/Minecolonies/minecolonies/blob/2c93afb9c00265d7894d87c94a983cd3b1ace2a6/src/main/java/com/minecolonies/colony/Colony.java#L659): Add parentheses to make the operator precedence explicit. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS864)

**Comment by cltnschlosser** [comment]
:shipit:

**Pull request merged**